server/accounts_test.go

```diff
@@ -106,6 +107,195 @@ func TestAccountIsolation(t *testing.T) {
 	}
 }
 
+func TestAccountIsolationExportImport(t *testing.T) {
+	getServer := func(t *testing.T) *Server {
+		t.Helper()
+
+		s := opTrustBasicSetup()
+		go s.Start()
+		if !s.ReadyForConnections(5 * time.Second) {
+			t.Fatal("failed to be ready for connections")
+		}
+		return s
+	}
+	getAccountConns := func(t *testing.T, s *Server, exp, imp string) (*nats.Conn, *nats.Conn) {
+		t.Helper()
+
+		buildMemAccResolver(s)
+		okp, err := nkeys.FromSeed(oSeed)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Setup exporter account.
+		accAPair, err := nkeys.CreateAccount()
+		if err != nil {
+			t.Fatal(err)
+		}
+		accAPub, err := accAPair.PublicKey()
+		if err != nil {
+			t.Fatal(err)
+		}
+		accAClaims := jwt.NewAccountClaims(accAPub)
+		if exp != "" {
+			accAClaims.Limits.WildcardExports = true
+			accAClaims.Exports.Add(&jwt.Export{
+				Name:    fmt.Sprintf("%s-stream-export", exp),
+				Subject: jwt.Subject(exp),
+				Type:    jwt.Stream,
+			})
+		}
+		accAJWT, err := accAClaims.Encode(okp)
+		if err != nil {
+			t.Fatal(err)
+		}
+		addAccountToMemResolver(s, accAPub, accAJWT)
+
+		// Setup importer account.
+		accBPair, err := nkeys.CreateAccount()
+		if err != nil {
+			t.Fatal(err)
+		}
+		accBPub, err := accBPair.PublicKey()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		accBClaims := jwt.NewAccountClaims(accBPub)
+		if imp != "" {
+			accBClaims.Imports.Add(&jwt.Import{
+				Name:    fmt.Sprintf("%s-stream-import", imp),
+				Subject: jwt.Subject(imp),
+				Account: accAPub,
+				Type:    jwt.Stream,
+			})
+		}
+		accBJWT, err := accBClaims.Encode(okp)
+		if err != nil {
+			t.Fatal(err)
+		}
+		addAccountToMemResolver(s, accBPub, accBJWT)
+
+		ncA := natsConnect(t, s.ClientURL(), createUserCreds(t, nil, accAPair),
+			nats.Name(fmt.Sprintf("nc-exporter-%s", exp)))
+		ncB := natsConnect(t, s.ClientURL(), createUserCreds(t, nil, accBPair),
+			nats.Name(fmt.Sprintf("nc-importer-%s", imp)))
+
+		return ncA, ncB
+	}
+
+	cases := []struct {
+		name     string
+		exp, imp string
+		pubSubj  string
+	}{
+		{
+			name: "export literal, import literal",
+			exp:  "foo", imp: "foo",
+			pubSubj: "foo",
+		},
+		{
+			name: "export full wildcard, import literal",
+			exp:  "foo.>", imp: "foo.bar",
+			pubSubj: "foo.bar",
+		},
+		{
+			name: "export full wildcard, import sublevel full wildcard",
+			exp:  "foo.>", imp: "foo.bar.>",
+			pubSubj: "foo.bar.whizz",
+		},
+		// FAIL - dropping "foo.bar" from different account.
+		// What is the correct behavior here?
+		// Should this be allowed?
+		// If so, probably need to edit accounts.go:2427
+		// {
+		// 	name: "export literal, import full wildcard",
+		// 	exp:  "foo.bar", imp: "foo.>",
```

Comment:
this looks like a violation imp should be a subset of exp, which this one wouldn't be